xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -1237,18 +1236,59 @@ where
         envelope: &GroupMessageV1,
         trust_message_order: bool,
     ) -> Result<MessageIdentifier, GroupMessageProcessingError> {
+        let mls_message_in = MlsMessageIn::tls_deserialize_exact(&envelope.data)?;
+        let message_entity_kind = match mls_message_in.wire_format() {
+            WireFormat::Welcome => EntityKind::Welcome,
+            _ => EntityKind::Group,
+        };
+
+        if trust_message_order {
```

Comment:
These `if trust_message_order` blocks can be removed when we have totally ordered streams.